launch_frontend/launch_frontend/parser.py

```diff
@@ -0,0 +1,113 @@
+# Copyright 2019 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Module for Parser class and parsing methods."""
+
+import io
+from typing import Text
+from typing import Union
+
+import launch
+from launch.utilities import is_a
+
+from pkg_resources import iter_entry_points
+
+from .entity import Entity
+from .expose import action_parse_methods
+from .parse_substitution import default_parse_substitution
+
+interpolation_fuctions = {
+    entry_point.name: entry_point.load()
+    for entry_point in iter_entry_points('launch_frontend.interpolate_substitution')
+}
+
+extensions_loaded = False
+
+
+class Parser:
+    """
+    Abstract class for parsing actions, substitutions and descriptions.
+
+    Implementations of the parser class, should override the load method.
+    They could also override the parse_substitution method, or not.
+    load_parser_extensions, parse_action and parse_description are not suposed to be overrided.
+    """
+
+    extensions_loaded = False
+    frontend_parsers = None
+
+    @classmethod
+    def load_parser_extensions(cls):
+        """Load parser extension, in order to get all the exposed substitutions and actions."""
+        if cls.extensions_loaded is False:
+            for entry_point in iter_entry_points('launch_frontend.parser_extension'):
+                entry_point.load()
```

Comment:
@ivanpauno meta: we can leave it as is for a first iteration, but I'm not entirely convinced that working with importation side effects is in our best interest in the long run. I wonder if having some sort of index at each package root module that its classes can expose/register themselves into would be cleaner than a having a global one. Maybe @wjwwood can weigh in too on this.

Reply:
It's not so much about performance (though I had not event considered that, thanks for the reference) but the way actions, substitutions and parsers are discovered. It's currently loading a global collection inside `launch` upon entry point loading. 

I was wondering if it wouldn't make sense to move away from that and make exposure somewhat more explicit, and maybe introducing namespaces in the process (which we do not have, we just fail upon tag name collision). Anyways, just a thought. No need to address this now, really.